src/movies.js

```diff
@@ -1,22 +1,86 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+
+      const newMovies = moviesArray.map(movie => movie.director);
+      return newMovies;
+  }
+
+   console.log(newMovies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+
+        if (moviesArray.length === 0) {
+            return 0;
+        }
+    const spielbergMovies = moviesArray.filter (movies => movies.director.includes ('Steven Spielberg') && movies.genre.includes ('Drama') === true);
+    return spielbergMovies.length;
+}
```

Comment:
Esto tambien esta perfect!! pero fijate que estas devolviendo cuando el length es 0, 0, pues devuelve el length directamente que es lo mismo! :D te lo refactorizo un poco!

```suggestion
function howManyMovies(moviesArray) {
    return moviesArray.filter (movies => movies.director.includes ('Steven Spielberg') && movies.genre.includes ('Drama')).length;
}
```

Reply:
asi se ve hasta mas claro la verdad, aunque claro, hacerlo ya es otra historia XD